src/SymbolPaths/QSettingsBasedStorageManager.cpp

```diff
@@ -36,4 +40,38 @@ void QSettingsBasedStorageManager::SavePaths(absl::Span<const std::filesystem::p
   settings.endArray();
 }
 
+void QSettingsBasedStorageManager::SaveModuleSymbolFileMappings(
+    const absl::flat_hash_map<std::string, std::filesystem::path>& mappings) {
+  QSettings settings{};
+  settings.beginWriteArray(kModuleSymbolFileMappingKey, static_cast<int>(mappings.size()));
+  int index = 0;
+  for (const auto& [module_path, symbol_file_path] : mappings) {
+    settings.setArrayIndex(index);
```

Comment:
I am going to disagree with you here, but I don't feel strongly about it. I think it's easier to read if the those 2 things (using index and incrementing index) are not together in one line, but in 2 lines. If it's in one line I need to think about whats happening, if its in 2, I can read it quickly. 

For a second I wanted to leave this PR open to discuss this, but I took a quick look at the the style guide https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement and it basically agrees with me here, so I am going to merge when the CI is done. The style guide says to use prefix increment whenever possible, and specifically "This [postfix increment] can result in code that is more compact but harder to read.". I think this is the case here.

Reply:
Perfect, consider that a nit.
In general I don't like to have ++index inside the body of a for, so I tend to include them in the for-loop. Here the usage it's so simple that I don't care to add it here. But if you feel that is confused and also have a negative comment in the cppguide, ignore my comment.

Reply:
In general I agree that having the index in the for loop is a better idea. But in this case that would be annoying here, because I need to iterate over the flat_hash_map mappings. (I would actually need to look up how that can be done). The `:` variant is just so much nicer for maps. In a perfect world the would be a built in way to get an index when iterating like that. Quite a few modern languages have that, but c++ of course not (I think its not even part of ranges in c++20👎)